lib/airbrake-ruby/code_hunk.rb

```diff
@@ -0,0 +1,57 @@
+module Airbrake
+  ##
+  # Represents a small hunk of code consisting of a base line and a couple lines
+  # around
+  class CodeHunk
+    ##
+    # @return [Integer] the maximum length of a line
+    MAX_LINE_LEN = 200
+
+    ##
+    # @return [Integer] how many lines should be read around the base line
+    NLINES = 2
+
+    def initialize(config, file, line, nlines = NLINES)
+      @config = config
+      @file = file
+      @line = line
+
+      @start_line = [line - nlines, 1].max
+      @end_line = line + nlines
+
+      @lines = {}
+    end
+
+    ##
+    # @return [Hash{Integer=>String}, nil] code hunk around the base line. When
+    #   an error occurrs, returns a zero key Hash
+    def to_h
+      return @lines if @lines.any?
+      return unless File.exist?(@file)
+
+      begin
+        fetch_code
```

Comment:
What I see is that you check `lines`, then for some reason call `fetch_code` and then check `lines` again. It would be much more obvious if code was:

```
check @lines
set @lines
check @lines
```

Reply:
Technically, checking lines on line 29 is not needed. It's just prevention of reading the same file twice (for the same results). I removed it for now.